allensdk/api/cloud_cache/README.md

```diff
@@ -0,0 +1,101 @@
+Cloud Cache
+===========
+
+## High level summary
+
+The classes defined in this directory are designed to provide programmatic
+access to version-controlled, cloud-hosted datasets. Users download these
+datasets using the `CloudCache` class defined in `cloud_cache.py`.
+The datasets accessed by `CloudCache` generally consist of three parts
+
+- Some arbitrary number of metadata files. These will be csv files suitable for
+reading with pandas.
+- Some arbitrary number of data files. These can be of any form.
+- A manifest.json file defining the contents of the dataset.
+
+For each version of the dataset, there will be a distinct manifest file loaded
+into the cloud service behind `CloudCache`. All other files are
+version-controlled using the cloud service's native functionality. To load a
+dataset, the user instantiates CloudCache and runs
+`cache.load_manifest('name_of_manifest.json')`. Valid manifest file names can
+be accessed through `cache.manifest_file_names`. Loading the manifest
+essentially configures `CloudCache` to access the corresponding version of the
+dataset.
+
+`CloudCache.data_path(file_id)` will download a data file to the local
+sytem and return the path to where that file has been downloaded. If the file
+has already been downloaded, `CloudCache.data_path(file_id)` will just return
+the path to the local copy of the file without downloading it again. In this
+call `file_id` is a unique identifier for each data file corresponding to a
+column in the metadata files. The name of that column can be found with
+`CloudCache.file_id_column`.
+
+`CloudCache.metadata_path(metadata_fname)` will download a metadata file to
+the local system and return the path where the file has been stored. The list
+of valid values for `metadata_fname` can be found with
+`CloudCache.metadata_file_names`. If users wish to directly access a
+pandas DataFrame of a given metadata file, they can use
+`CloudCache.metadata(metadata_fname)`.
+
+## Structure of `manifest.json`
+
+The `manifest.json` files are structured like so
+```
+
+{
+ "dataset_version" : dataset_version_string,
+ "file_id_column": name_of_column_uniquely_identifying_files,
+ "metadata_files":{
+     metadata_file_name_1: {"uri": "full/uri/to/file",
+                            "version_id": version_id_string,
+                            "file_hash": file_hash_of_metadata_file},
+     metadata_file_name_2: {"uri": "full/uri/to/file",
+                            "version_id": version_id_string,
+                            "file_hash": file_hash_of_metadata_file},
+  ...
+ },
+ "data_files": {
+     file_id_1: {"uri": "full/uri/to/imaging_plane.nwb",
```

Comment:
I think Tim or Rob mentioned that these are technically `url`'s (URL) not `uri` (URI).

Reply:
Fixed